### PR TITLE
[Backport][SES5] inital apparmor support

### PR DESCRIPTION
**Pull request by jan--f**
Have yet to test things.

### CODE REVIEW

File: srv/modules/runners/ready.py [hunk 1]
```diff
@@ -71,17 +71,39 @@ def apparmor(self):
...
-                else:
-                    self.warnings['apparmor'] = [ msg ]
-        if 'apparmor' not in self.warnings:
+                minions_with_apparmor += 1
+                # check if all file.managed for the profiles would apply
+                # cleanly using test=True. If so salt returns 'result': True
+                # burried in its return structure
+                result = self.local.cmd(minion, 'state.apply',
+                                        ['ceph.apparmor.profiles'],
+                                        kwarg={'test': 'true'})
```
Comment (swiftgist) [inline srv/modules/runners/ready.py]: I'm worried about calling minions sequentially.   Calling once in parallel we keep the runtime more consistent.  (We have this in #1104 too.)

File: srv/salt/ceph/smoketests/apparmor/init.sls [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{% set master = salt['master.minion']() %}
```
Comment (jschmid1) [inline srv/salt/ceph/smoketests/apparmor/init.sls]: That needs to be converted to master_minion

**Comment by jschmid1** [comment]
TemplateNotFound: ceph/macros/os_switch.sls

That was likely forgot in the backports

**Comment by jan--f** [comment]
k, I added the os_switch commits to this backport. It makes sense to have those here since the apparmor states are the only ones using that switch as of yet.

File: srv/modules/runners/ready.py (hunk 1 above)
Comment (jan--f) [reply srv/modules/runners/ready.py]: fair. Fixed in the PR to master...will update backport.

**Comment by jan--f** [comment]
alright...comments are addressed.

**Comment by jschmid1** [comment]
This time:
````

   data1.ceph:                                                                                                                      
    ----------                                                                                                                       
              ID: patterns-base-apparmor                                                                                             
        Function: pkg.installed                                                                                                      
          Result: False                                                                                                              
         Comment: An error was encountered while installing package(s): Zypper command failure: Package 'patterns-base-apparmor' not found.
         Started: 11:17:40.777608                                                                                                    
        Duration: 4733.987 ms                                                                                                        
         Changes:    
````


`zypper -s patterns-base-apparmor`

does not yield any results.

There is `patterns-sles-apparmor` though

**Comment by jan--f** [comment]
Smoketests work fine for me in a fresh sp3 cluster with ceph running:
```
 salt-run state.orch ceph.smoketests.apparmor
[WARNING ] Unable to find IPv6 record for "admin" causing a 10 second timeout when rendering grains. Set the dns or /etc/hosts for IPv6 to clear this.
[WARNING ] Unable to find IPv6 record for "admin" causing a 10 second timeout when rendering grains. Set the dns or /etc/hosts for IPv6 to clear this.
admin_master:
  Name: enforce apparmor profiles - Function: salt.state - Result: Changed Started: - 07:33:21.450160 Duration: 45742.948 ms
  Name: make sure ceph cluster is healthy - Function: salt.state - Result: Changed Started: - 07:34:07.193413 Duration: 6480.125 ms

Summary for admin_master
------------
Succeeded: 2 (changed=2)
Failed:    0
------------
Total states run:     2
Total run time:  52.223 s
```

**Pull request merged**